studio/components/interfaces/SQLEditor/RenameQueryModal.tsx

```diff
@@ -16,10 +16,15 @@ export interface RenameQueryModalProps {
 
 const RenameQueryModal = ({ snippet, visible, onCancel, onComplete }: RenameQueryModalProps) => {
   const { ui } = useStore()
-  const { ref } = useParams()
+  const { ref, slug } = useParams()
   const snap = useSqlEditorStateSnapshot()
   const supabaseAIEnabled = useFlag('sqlEditorSupabaseAI')
 
+  const { data: subscription } = useOrgSubscriptionQuery({ orgSlug: slug })
+
+  // Customers on HIPAA plans should not have access to Supabase AI
+  const isHipaaSubscription = subscription?.plan?.id === 'hipaa'
```

Comment:
HIPAA is an add-on, not an individual plan, here's a sample response, stripped down to the important bits from the subscription (same response for org and project endpoints)

```json
{
    "plan": {
        "id": "pro",
        "name": "Pro",
        "price": 25
    },
    "addons": [
        {
            "name": "HIPAA",
            "supabase_prod_id": "addon_security_hipaa",
            "price": 299
        }
    ]
}
```

So the proper way to check it would be 

`subscriptions.addons.some(it => it.supabase_prod_id === 'addon_security_hipaa')`